analyses/molecular-subtyping-CRANIO/00-craniopharyngiomas-molecular-subtype.Rmd

```diff
@@ -0,0 +1,180 @@
+---
+title: "Molecularly Subtype Craniopharyngiomas into Adamantinomatous or Papillary"
+author: "Daniel Miller for D3B <millerd15@email.chop.edu>"
+date: "October 2020"
+output:
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+---
+
+# Background
+
+This notebook looks at the defining lesions for all samples for the issue of
+molecular subtyping craniopharyngiomas into adamantinomatous or papillary samples
+in the OpenPBTA dataset.
+
+Defining Lesions:
+
+CRANIO, ADAM
+- Contains SNV in exon 3 of CTNNB1
+- Tumor occurs mostly in childhood or young adolescence (0-39 years), but can be seen in adults
+
+CRANIO, PAP
+- Contains BRAF V600E mutation
+- Tumor occurs exclusively in adults (40+ years)
+
+# Usage
+
+This notebook is intended to be run via the command line from the top directory
+of the repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/molecular-subtyping-CRANIO/00-craniopharyngiomas-molecular-subtype.Rmd', clean = TRUE)"
+```
+
+# Set up
+
+```{r load-library}
+library(tidyverse)
+```
+
+## Directories and File Inputs/Outputs
+
+```{r set-dir-and-file-names}
+# Detect the ".git" folder -- this will in the project root directory.
+# Use this as the root directory to ensure proper sourcing of functions no
+# matter where this is called from
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+# File path to results directory
+results_dir <-
+  file.path(root_dir, "analyses", "molecular-subtyping-CRANIO", "results")
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+# Outputs
+output_lesions <- file.path(results_dir, "CRANIO_defining_lesions.tsv")
+output_subtype <- file.path(results_dir, "CRANIO_molecular_subtype.tsv")
+```
+
+```{r load-inputs-please-wait}
+# Inputs
+## Read in metadata
+metadata <-
+  read.table(file.path(root_dir, "data", "pbta-histologies.tsv"),
```

Comment:
I think if you use `readr::read_tsv()` function here instead of `read.table()` you may not have to do the `as.numeric(as.character(age_at_diagnosis_days))` coercing later. `read.table()` doesn't generally do a great job determining what kind of variables things are. (Things like `stringsAsFactor = FALSE` argument can help, but you could also just use `readr::read_tsv()` which generally does a better job with this and then you don't need your other `header` or `sep` arguments either.

Reply:
Very helpful. Seems like read.table was dropping rows. It only collected 38 observations whereas read_tsv collected 74.